### PR TITLE
Implement Multiple User Data (Typescript)

### PR DESCRIPTION
**Pull request by WilliamhGDP**
## Asana Task
- https://app.asana.com/0/1206389362636911/1207995538212471/f

## Description
- Add multiple user data on cli, client and issuer-verifier (node-js)

### Cli
1. go to folder `typescript/node-js/cli`
2. run `npm i`
3. run `npm run demo-multi`

### Issuer-verifier & client
1. go to folder `typescript/node-js/client` & `typescript/node-js/issuer-verifier`
2. run `npm i` on both folder
3. run `npm run dev` on both folder
4. login as any of the user ("john", "jane", etc)
5. click on `start employee onboarding (single user data)` and `start employee onboarding (multiple user data)` buttons

Scenario:
  1. john:
      - single: pass
      - multiple: pass
  2. jake: 
      - single: pass
      - multiple: not pass (mismatch kycType)
  3. jane:
      - single: not pass (high cocaine)
      - multiple: not pass (high cocaine)
  4. rob:
      - single: pass
      - multiple: not pass (mismatch dateOfBirth)
  5. kyle:
      - single: not pass (high cocaine)
      - multiple: not pass (high cocaine & mismactch kycId)

### CODE REVIEW

File: typescript/node-js/issuer-verifier/src/utils/constants.ts [hunk 1]
```diff
@@ -5,8 +5,8 @@
  *   NaufalFakhri (naufal.f.muhammad@gdplabs.id)
  * Created at: December 15th 2023
  * -----
- * Last Modified: May 17th 2024
- * Modified By: LawrencePatrickSianto (lawrence.p.sianto@gdplabs.id)
+ * Last Modified: August 20th 2024
+ * Modified By: William H Hendrawan (william.h.hendrawan@gdplabs.id)
```
Comment (khandar-william) [inline typescript/node-js/issuer-verifier/src/utils/constants.ts:9]: PII dihapus atau tidak?

File: typescript/node-js/issuer-verifier/src/utils/signing.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/*
+ * signing.ts
+ *
+ * Authors:
+ *   William H Hendrawan (william.h.hendrawan@gdplabs.id)
+ * Created Date: August 20th 2024
+ * -----
+ * Last Modified: August 20th 2024
+ * Modified By: William H Hendrawan (william.h.hendrawan@gdplabs.id)
```
Comment (khandar-william) [inline typescript/node-js/issuer-verifier/src/utils/signing.ts:9]: PII dihapus atau tidak?

File: typescript/node-js/issuer-verifier/src/app/verifier/dvrs/dvrHelper.ts [hunk 3]
```diff
@@ -2,17 +2,6 @@
  * dvrHelper.ts
  * DEFAULT DESCRIPTION. EDIT OR DELETE THIS.
```
Comment (khandar-william) [inline typescript/node-js/issuer-verifier/src/app/verifier/dvrs/dvrHelper.ts:3]: delete line

File: typescript/node-js/issuer-verifier/src/app/verifier/dvrs/route.ts [hunk 4]
```diff
@@ -6,8 +6,8 @@
  *   Zulchaidir (zulchaidir@gdplabs.id)
  * Created at: October 31st 2023
  * -----
- * Last Modified: July 27th 2024
- * Modified By: LawrencePatrickSianto (lawrence.p.sianto@gdplabs.id)
+ * Last Modified: August 20th 2024
+ * Modified By: William H Hendrawan (william.h.hendrawan@gdplabs.id)
```
Comment (khandar-william) [inline typescript/node-js/issuer-verifier/src/app/verifier/dvrs/route.ts:10]: PII dihapus ga?

**Comment by janice-laksana** [comment]
> ## Asana Task
> 
>     * https://app.asana.com/0/1206389362636911/1207995538212471/f
> 
> 
> ## Description
> 
>     * Add multiple user data on cli, client and issuer-verifier (node-js)
> 
> 
> ## How to test
> ### Cli
> 
>     1. go to folder `typescript/node-js/cli`
> 
>     2. run `npm i`
> 
>     3. run `npm run test` to make sure nothing is broken
> 
>     4. run `npm run demo-multi`
> 
> 
> ### Issuer-verifier & client
> 
>     1. go to folder `typescript/node-js/client` & `typescript/node-js/issuer-verifier`
> 
>     2. run `npm i` on both folder
> 
>     3. run `npm run dev` on both folder
> 
>     4. login as any of the user ("john", "jane", etc)
> 
>     5. click on `start employee onboarding (single user data)` and `start employee onboarding (multiple user data)` buttons
> 
> 
> Scenario:
> 
>     1. john:
>        
>        * single: pass
>        * multiple: pass
> 
>     2. jake:
>        
>        * single: pass
>        * multiple: not pass (mismatch kycType)
> 
>     3. jane:
>        
>        * single: not pass (high cocaine)
>        * multiple: not pass (high cocaine)
> 
>     4. rob:
>        
>        * single: pass
>        * multiple: not pass (mismatch dateOfBirth)
> 
>     5. kyle:
>        
>        * single: not pass (high cocaine)
>        * multiple: not pass (high cocaine & mismactch kycId)

ini npm run testnya gk bisaa We... hapus aja dari description ini..

**Comment by janice-laksana** [comment]
dari aku,, tinggal solved komen Khanwil... terus oke sih

**Review by janice-laksana — APPROVED** [review]

File: typescript/node-js/issuer-verifier/src/utils/constants.ts (hunk 1 above)
Comment (WilliamhGDP) [reply typescript/node-js/issuer-verifier/src/utils/constants.ts]: https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/2ee77fc53bb6d7c4307daa23c3da9f0f2ecd56fd

File: typescript/node-js/issuer-verifier/src/utils/signing.ts (hunk 2 above)
Comment (WilliamhGDP) [reply typescript/node-js/issuer-verifier/src/utils/signing.ts]: done https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/2ee77fc53bb6d7c4307daa23c3da9f0f2ecd56fd

File: typescript/node-js/issuer-verifier/src/app/verifier/dvrs/dvrHelper.ts (hunk 3 above)
Comment (WilliamhGDP) [reply typescript/node-js/issuer-verifier/src/app/verifier/dvrs/dvrHelper.ts]: done https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/2ee77fc53bb6d7c4307daa23c3da9f0f2ecd56fd

File: typescript/node-js/issuer-verifier/src/app/verifier/dvrs/route.ts (hunk 4 above)
Comment (WilliamhGDP) [reply typescript/node-js/issuer-verifier/src/app/verifier/dvrs/route.ts]: done https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/2ee77fc53bb6d7c4307daa23c3da9f0f2ecd56fd

File: typescript/node-js/client/src/app/users/usersWelcome.component.tsx [hunk 5]
```diff
@@ -38,8 +28,11 @@ export default function UsersWelcome({ user }: { user: string | undefined }) {
           elevation={3}
           className='w-2/5 p-7 flex flex-col items-center gap-5'>
           <div>Welcome {user}!</div>
-          <Button href='/onboarding' variant='outlined'>
-            Start Employee Onboarding
+          <Button href='/onboarding?user-data=single' variant='outlined'>
+            Start Employee Onboarding {"(Single User Data)"}
+          </Button>
+          <Button href='/onboarding?user-data=multiple' variant='outlined'>
```
Comment (LawrencePatrickSianto) [inline typescript/node-js/client/src/app/users/usersWelcome.component.tsx:34]: Tambahi `className='text-center'` di button supaya kalau window browser kecil, tulisannya centered di tengah

**Review by LawrencePatrickSianto — COMMENTED** [review]
Setelah proof verified (failed/success), sebaiknya ditambah button `Start Over` atau semacamnya agar bisa balik ke menu milih Single/Multiple User Data. 

Kalau sekarang, kita mesti logout dan login lagi buat coba ulang user yang sama.

**Review by khandar-william — APPROVED** [review]
yg kali ini gw ga sempet tes jalanin, tapi ngeliat kodenya ok saja

File: typescript/node-js/client/src/app/users/usersWelcome.component.tsx (hunk 5 above)
Comment (WilliamhGDP) [reply typescript/node-js/client/src/app/users/usersWelcome.component.tsx]: done https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/18b67539920064d0965b1a8551f357293850313e

Comment (LawrencePatrickSianto) [reply typescript/node-js/client/src/app/users/usersWelcome.component.tsx]: yang button `Back to home`, cuma muncul kalo proofnya success, kalau proof gagal ga muncul. Expectku muncul dua-duanya.

Terus mending Button nya outlined, kalau cuma text dia bisa ngira itu bukan button, cuma text.

Tulisannya ubah dari `Bach to home` jadi `Back to Home`

Comment (WilliamhGDP) [reply typescript/node-js/client/src/app/users/usersWelcome.component.tsx]: done, salah taruh
https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/4643c8a9ce72783ae29eadbbaa882efd4e40d9e3
https://github.com/gl-zkPass/zkpass-sdk/pull/79/commits/532edeac476992a64ef9b88a06a908fa3f4eff59

**Review by LawrencePatrickSianto — APPROVED** [review]

**Pull request merged**